src/waitFor.ts

```diff
@@ -198,6 +191,17 @@ export default async function waitFor<T>(
     return waitForInternal(expectation, optionsWithStackTrace);
   }
 
+  if (checkReactVersionAtLeast(18, 0)) {
```

Comment:
nit:style:

Pls put this `if` above the `if (!checkReactVersionAtLeast(16, 9)) {` in order to improve readability